ui/desktop/electron-app/config/cli.js

```diff
@@ -66,13 +66,31 @@ const downloadArtifact = (version) => {
   });
 };
 
-const extract = (artifactPath, destination) => {
+const extract = async (artifactPath, destination) => {
   if (!fs.existsSync(destination)) fs.mkdirSync(destination);
+  const directory = await unzipper.Open.file(artifactPath);
 
-  const zip = new admZip(artifactPath);
-  // The first boolean flag is to overwrite files, the second is to preserve original
-  // file permissions which doesn't seem to be documented.
-  return zip.extractAllTo(destination, true, true);
+  return Promise.all(
+    directory.files.map((file) => {
+      return new Promise((resolve, reject) => {
+        file
+          .stream()
+          .pipe(
+            fs.createWriteStream(path.join(destination, file.path), {
+              // Creating a new file stream sets the permission bits on the file
+              // to node's default. We need to set the permission bits ourselves.
+              // We could just force the mode to 0o755, but we can also just convert it.
+              // This should also work on windows as the permission bits will still be set
+              // correctly but they just won't have any effect.
+              // Taken from here: https://github.com/thejoshwolfe/yauzl/issues/101#issuecomment-448073570
+              mode: file.externalFileAttributes >>> 16,
```

Comment:
clarifying question: so if we shift the `externalFileAttributes` field by 16 the result will always be the binary version of `0o755`?

Reply:
Not necessarily, it depends on the original externalFileAttributes. The issue was the `externalFileAttributes` wasn't in a format that could be passed into mode directly (I think it's encoded?) to set the permissions, and this was to mainly convert it.

The original file permissions for the boundary executable was `755` which it ended up converting into but if you notice in the screenshot, the license file ended up as `644` (I'm not sure if it ends up as `644` bc it was invalid or if that was the original permissions but we really only care about the CLI).

I could have also alternatively only extracted out the CLI but I would then have to get the right name depending if we're on windows or not but that didn't seem necessary

Reply:
Thank you for the explanation, that helped a lot!